src/engraving/draw/painterpath.cpp

```diff
@@ -0,0 +1,751 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "painterpath.h"
+
+#include "global/log.h"
+
+namespace mu {
+
+static constexpr double pi = 3.14159265358979323846;
+static constexpr double pathKappa = 0.5522847498;
+
+static void findEllipseCoords(const RectF &r, double angle, double length,
+                            PointF* startPoint, PointF *endPoint);
+
+static PointF curvesForArc(const RectF &rect, double startAngle, double sweepLength,
+                          PointF *curves, int *point_count);
+
+static double angleForArc(double angle);
+
+void PainterPath::moveTo(const PointF& p)
+{
+    if (!hasValidCoords(p)) {
+#ifdef DEBUG
+        LOGW() << "PainterPath::moveTo: Adding point with invalid coordinates, ignoring call");
+#endif
```

Comment:
In some places for debugging we use the `TRACE_DRAW_OBJ_ENABLED` definition
Better to use it.
Otherwise, there will be a lot of messages in the debug that will interfere.

Reply:
I think it's better to also rename `TRACE_DRAW_OBJ_ENABLED` to `TRACE_DRAW_ENABLED`